src/test/java/edu/hm/hafner/util/SetTest.java

```diff
@@ -0,0 +1,176 @@
+package edu.hm.hafner.util;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.*;
+
+import java.util.*;
+import java.util.HashSet;
+
+/**
+ * Tests the class HashSet.
+ *
+ * @author Sofia Steinleitner
+ */
+class SetTest {
+    private static final Integer eins = 1;
+    private static final Integer zwei = 2;
+    private static final Integer drei = 3;
+    private static final Integer vier = 4;
+
+    @Test
+    void shouldTestConstructorHashSet() {
+        HashSet<Integer> set = new HashSet<Integer>();
+        assertThat(set).isEmpty();
+    }
+
+    @Test
+    void shouldTestConstructorCollection() {
+        assertThatThrownBy(() -> new HashSet<Integer>(null))
+                .isExactlyInstanceOf(NullPointerException.class)
+                .as("If the specified collection is null, throws NullPointerException");
```

Comment:
Haben Sie mal überprüft, ob da die gewünschte Fehlermeldung kommt? Meines Wissens muss das `as` vor dem Check stehen, also:
```suggestion
        assertThatThrownBy(() -> new HashSet<Integer>(null))
                .as("If the specified collection is null, throws NullPointerException")
                 .isExactlyInstanceOf(NullPointerException.class);
```